File: vipclaw-common-entity/src/main/java/com/vipamp/vipclaw/common/entity/ToolCallLogEntity.java
package com.vipamp.vipclaw.common.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.io.Serial;
import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 工具调用日志实体类
 *
 * @author vipamp
 * @since 2026-04-14
 */
@Data
@EqualsAndHashCode(callSuper = false)
@TableName("tool_call_log")
@Schema(description = "工具调用日志实体类")
public class ToolCallLogEntity implements Serializable {

    @Serial
    private static final long serialVersionUID = 1L;

    /**
     * ID
     */
    @TableId(value = "id", type = IdType.AUTO)
    @Schema(description = "ID")
    private Long id;

    /**
     * 智能体 ID
     */
    @Schema(description = "智能体 ID")
    private Long agentId;

    /**
     * 会话 ID
     */
    @Schema(description = "会话 ID")
    private String sessionId;

    /**
     * 工具名称
     */
    @Schema(description = "工具名称")
    private String toolName;

    /**
     * 工具参数（JSON 格式）
     */
    @Schema(description = "工具参数（JSON 格式）")
    private String args;

    /**
     * 工具执行结果
     */
    @Schema(description = "工具执行结果")
    private String result;

    /**
     * 是否成功（1-成功，0-失败）
     */
    @Schema(description = "是否成功（1-成功，0-失败）")
    private Integer success;

    /**
     * 开始时间戳
     */
    @Schema(description = "开始时间戳")
    private LocalDateTime startTime;

    /**
     * 结束时间戳
     */
    @Schema(description = "结束时间戳")
    private LocalDateTime endTime;

    /**
     * 执行耗时（毫秒）
     */
    @Schema(description = "执行耗时（毫秒）")
    private Long duration;

    /**
     * 时间戳
     */
    @Schema(description = "时间戳")
    private LocalDateTime ts;
}

File: vipclaw-admin/src/main/kotlin/com/vipamp/vipclaw/admin/entity/ToolCallLogEntity.kt
package com.vipamp.vipclaw.admin.entity

import io.swagger.v3.oas.annotations.media.Schema
import java.io.Serializable
import java.time.LocalDateTime

@Schema(description = "Tool call log entity")
class ToolCallLogEntity : Serializable {
    companion object {
        private const val serialVersionUID = 1L
    }

    @Schema(description = "ID")
    var id: Long = 0

    @Schema(description = "Agent ID")
    var agentId: Long = 0

    @Schema(description = "Session ID")
    var sessionId: String = ""

    @Schema(description = "Tool name")
    var toolName: String = ""

    @Schema(description = "Tool arguments (JSON format)")
    var args: String = ""

    @Schema(description = "Tool execution result")
    var result: String = ""

    @Schema(description = "Success status (1-success, 0-failure)")
    var success: Int = 0

    @Schema(description = "Start timestamp")
    var startTime: LocalDateTime = LocalDateTime.now()

    @Schema(description = "End timestamp")
    var endTime: LocalDateTime = LocalDateTime.now()

    @Schema(description = "Execution duration (milliseconds)")
    var duration: Long = 0

    @Schema(description = "Timestamp")
    var ts: LocalDateTime = LocalDateTime.now()
}
